opts.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "USAGE\n")
+	fmt.Fprintf(os.Stderr, "  kubernetes-ipfs [--param=<name>:<value>,...] <testfile>\n\n")
+	fmt.Fprintf(os.Stderr, "OPTIONS\n")
+	// print each flag's description
+	flag.PrintDefaults()
+	// tack on `--help` flag at the end, previous command doesn't print it
+	fmt.Fprintf(os.Stderr,
+		`  -help
+        Show this help message and exit`)
+	fmt.Fprintf(os.Stderr, "\n")
+}
+
+// Opts holds the result of parsed input args
+type Opts struct {
+	params Params
+}
+
+// all input options + corresponding data structures should be defined here
+func newOpts() *Opts {
+	opts := &Opts{
+		params: make(Params),
+	}
+
+	flag.Var(&opts.params, "param",
+		`Replace all test parameter instances of <name> with <value>.
+        Separate multiple `+"`<name>:<value>` inputs with ',' or pass this flag multiple times.")
+
+	return opts
+}
+
+/* parse all input args */
+func parseArgs() (*Opts, []string) {
+	// create opts
+	opts := newOpts()
+	// set usage
+	flag.Usage = usage
+	// parse all args
+	flag.Parse()
+
+	// get all non-flag args
+	rest := append([]string{os.Args[0]}, flag.Args()...)
+
+	return opts, rest
+}
+
+// Params holds the parameter mappings specified  by user
+type Params map[PName]PValue
+type PName string
+type PValue string
+
+var pNameRegexString = `([A-Za-z_][0-9A-Za-z_]*)`
+var pNameRegex = regexp.MustCompile(`^` + pNameRegexString + `$`)
+var paramRegex = regexp.MustCompile(`{{ ` + pNameRegexString + ` }}`)
```

Comment:
Should also match if I write `{{myvar}}` or `{{   myvar}}` or `{{    myvar               }}`. imho. Should ignore spaces.